### PR TITLE
Adds custom message format.

### PR DESCRIPTION
**Pull request by tavilarion**
## Pull Request Checklist

**Is this in reference to an existing issue?**

#### General

- [ ] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

- [ ] Update README with any necessary configuration snippets

- [ ] Binstubs are created if needed

- [ ] RuboCop passes

- [ ] Existing tests pass

#### New Plugins

- [ ] Tests

- [ ] Add the plugin to the README

- [ ] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

### CODE REVIEW

**Comment by majormoses** [comment]
@eheydrick ping since I know you use opsgenie.

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,12 @@ This CHANGELOG follows the format listed at [Keep A Changelog](http://keepachang
 
 ## [Unreleased]
 
+## [3.1.0] - 2017-11-07
```
Comment (majormoses) [inline CHANGELOG.md]: as there is no guarantee when we get to PRs or the order that they are merged we accept contributions as [Unreleased change](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md#unreleased-change) and leave the versioning to the maintainers this allows us to also validate that the contribution is versioned correctly from a semver perspective.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,12 @@ This CHANGELOG follows the format listed at [Keep A Changelog](http://keepachang
...
+### Added
+- Added `-f, --format` parameter to be able to specify a custom message format
```
Comment (majormoses) [inline CHANGELOG.md]: Would it make sense to instead provide a template of it? I am not sure how complex things might become with the formatting. Might me nice to specify an erb, jinja, etc style template and then just pass it the location of the template.

File: bin/handler-opsgenie.rb [hunk 3]
```diff
@@ -10,37 +10,72 @@
...
+  def message
+    return @event['notification'] unless @event['notification'].nil?
+    return default_message if message_format.nil?
+    custom_message
+  rescue StandardError
+    default_message
+  end
+
+  def custom_message
+    message_format.gsub(/\<{2}\w+\>{2}/) do |value|
```
Comment (majormoses) [inline bin/handler-opsgenie.rb]: same comment regarding using an erb, jinja, etc template.

**Comment by majormoses** [comment]
Thanks for your contribution to Sensu plugins! Without people like you submitting PRs we couldn't run the project. I will review it shortly.

File: CHANGELOG.md (hunk 2 above)
Comment (tavilarion) [reply CHANGELOG.md]: Sounds like a good idea, will come up with a solution later today.

**Comment by tavilarion** [comment]
Changed the option to receive the path to a ERB template, please let me know what you think.

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,10 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 This CHANGELOG follows the format listed at [Keep A Changelog](http://keepachangelog.com/)
 
 ## [Unreleased]
+### Add custom message template
```
Comment (majormoses) [inline CHANGELOG.md]: This should just by the type of change in this case `Added` as it is a new feature.

**Review by majormoses — CHANGES_REQUESTED** [review]
Just fix up the changelog and a review from @eheydrick and we should be good to go.

**Comment by tavilarion** [comment]
Fixed the changelog.

**Review by majormoses — APPROVED** [review]

**Comment by majormoses** [comment]
will give it a day and see if @eheydrick has any thoughts since he uses this.

**Comment by tavilarion** [comment]
Thanks for the prompt response guys 👍

**Comment by majormoses** [comment]
Alright looks like we are gonna just merge this. @eheydrick whenever you are back you should probably retroactively review.

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-opsgenie/versions/3.1.0

**Pull request merged**